ytfzf

```diff
@@ -1031,186 +1033,148 @@ bad_opt_arg () {
+			scrape="yt_subs" ;;
+		subs)
+			sub_link_count=$optarg
+			is_non_number "$sub_link_count" && bad_opt_arg "--subs=" "$sub_link_count"
+			parse_opt "S" ;;
 
-		T)	
-		    trending_tab=${optarg:-}
-		    scrape="trending" ;;
+		fancy-subs)
+			fancy_subscriptions_menu=${optarg:-1}
+			is_non_number "$fancy_subscriptions_menu" && bad_opt_arg "--fancy-subs=" "$fancy_subscriptions_menu" ;;
+			
+		T|trending)	
+			trending_tab=${optarg:-}
+			scrape="trending" ;;
 
-		l) 	enable_loop=${optarg:-1} ;;
+		l|loop)
+			enable_loop=${optarg:-1}
+			is_non_number "$YTFZF_LOOP" && bad_opt_arg "--loop=" "$enable_loop" ;;
 
-		t) 	show_thumbnails=${optarg:-1} ;;
+		t|show-thumbnails)
+			show_thumbnails=${optarg:-1}
+			is_non_number "$show_thumbnails" && bad_opt_arg "--show-thumbnails=" "$show_thumbnails" ;;
+
+		thumbnail-quality)
+			thumbnail_quality=$optarg
+			is_non_number "$thumbnail_quality" && bad_opt_arg "--thumbnail-quality=" "$thumbnail_quality"
+			parse_opt "t" ;;
 
 		v)	printf "ytfzf: %s\n" "$YTFZF_VERSION"
 			exit ;;
+		version)
+			printf "\033[1mytfzf:\033[0m %s\n" "$YTFZF_VERSION"
+			printf "\033[1myoutube-dl:\033[0m %s\n" "$(youtube-dl --version)"
+			command -v "fzf" 1>/dev/null && printf "\033[1mfzf:\033[0m %s\n" "$(fzf --version)"
+			exit ;;
+
+		subt)
+			auto_caption=${optarg:-1}
+			is_non_number "$auto_caption" && bad_opt_arg "--subt=" "$auto_caption" ;;
 
-		L) 	show_link_only=${optarg:-1} ;;
+		L|show-link-only)
+			show_link_only=${optarg:-1}
+			is_non_number "$show_link_only" && bad_opt_arg "--show-link-only=" "$show_link_only" ;;
 
-		n)
-		    link_count="$optarg"
-		    is_non_number "$link_count" && bad_opt_arg "-n" "$link_count" ;;
+		n|link-count)
+			link_count="$optarg"
+			is_non_number "$link_count" && bad_opt_arg "-n" "$link_count" ;;
 
 		U) 	[ -p "$FIFO" ] && preview_img "$optarg"; exit;
 			# This option is reserved for the script, to show image previews
 			# Not to be used explicitly
 			;;
 
-		N)	enable_noti=${optarg:-1} ;;
+		N|notification)
+			enable_noti=${optarg:-1}
+			is_non_number "$enable_noti" && bad_opt_arg "--notification=" "$enable_noti" ;;
+
+		add-subs) create_subs ;;
+
+		upload-time) upload_date_filter=$optarg ;;
+		last-hour|today|this-week|this-month|this-year) upload_date_filter="$opt" ;;
+
+		upload-sort) sort_by_filter=$optarg ;;
+		upload-date|view-count|rating) sort_by_filter=$opt ;;
+
+		filter-id|sp) sp=$optarg ;;
+
+		preview-side) export PREVIEW_SIDE=$optarg ;;
+
+		update) update_ytfzf "master" ;;
+		update-unstable) update_ytfzf "${optarg-development}" ;;
 
 		*)
-			usageinfo
-			exit 2 ;;
+			if [ "$exit_on_opt_error" -eq 1 ]; then
+			    usageinfo
+			    exit 2
+			fi ;;
 	esac
 	unset opt optarg
 }
 
 
 while getopts "LhDmdfxHaArltSsvNTn:U:-:" OPT; do
+	# to parse extra-options in conf.sh
+	[ "$OPT" = "-" ] && OPT="${OPTARG%%=*}" && OPTARG="${OPTARG#*=}"
+	#when there is no = in OPTARG and it's a longopt, OPTARG will = OPT
+	[ "$OPTARG" = "$OPT" ] && OPTARG=""
```

Comment:
Since this part is only when `$OPT` is `-`. It would be a lil tidy to have an `if` statement here, for both these lines